cli/pkg/github/git.go

```diff
@@ -0,0 +1,47 @@
+package github
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+)
+
+func PublicKey() (*ssh.PublicKeys, error) {
+	var publicKey *ssh.PublicKeys
+	sshPath := os.Getenv("HOME") + "/.ssh/id_ed25519"
+	sshKey, _ := os.ReadFile(sshPath)
+	publicKey, err := ssh.NewPublicKeys("git", sshKey, "")
+	if err != nil {
+		return nil, err
+	}
+	return publicKey, err
+}
+
+func CloneRepo(url string) (string, error) {
+	repoName := strings.TrimSuffix(url[strings.LastIndex(url, "/")+1:], ".git")
```

Comment:
Use a proper URL parser for this – looks like there's one for git URLs in `go-git/plumbing/transport.NewEndpoint`

Reply:
Actually might not be needed per comment below about not doing special handling for github

Reply:
`transport.NewEndpoint` don't have option of getting repo name, so parsing it here. Other details we can get it via above URL parser